consensus/core/src/synchronizer.rs

```diff
@@ -36,12 +36,142 @@ const FETCH_FROM_PEERS_TIMEOUT: Duration = Duration::from_millis(4_000);
 
 const MAX_FETCH_BLOCKS_PER_REQUEST: usize = 200;
 
+const MAX_AUTHORITIES_TO_FETCH_PER_BLOCK: usize = 2;
+
+struct BlocksGuard {
```

Comment:
I've introduced the guard to allow us conveniently hold on the locks by just passing around the guard instead of having to manually call the inflight blocks map and unlock the blocks. I also consider it robust especially for cases where methods need to early return/error and resources automatically freed up without running the risk of keeping them locked for ever.